packages/core/src/domain/session/sessionStore.ts

```diff
@@ -166,7 +178,14 @@ export function startSessionStore<TrackingType extends string>(
   }
 
   function isSessionInCacheOutdated(sessionState: SessionState) {
-    return sessionCache.id !== sessionState.id || sessionCache[productKey] !== sessionState[productKey]
+    const didSessionIdChange = sessionCache.id !== sessionState.id
+    const didSessionTrackingChange = sessionCache[productKey] !== sessionState[productKey]
+    const untoleratedTrackingChange =
+      (!allowedTrackingTransition && didSessionTrackingChange) ||
+      (allowedTrackingTransition && sessionState[productKey] !== allowedTrackingTransition && didSessionTrackingChange)
+
+    const a = didSessionIdChange || untoleratedTrackingChange
+    return a
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏ 
```suggestion
    return didSessionIdChange || untoleratedTrackingChange
```